oak_functions/loader/src/server.rs

```diff
@@ -145,13 +144,25 @@ pub trait OakApiNativeExtension {
     /// Performs any cleanup or terminating behavior necessary before destroying the WasmState.
     fn terminate(&mut self) -> anyhow::Result<()>;
 }
+
 pub trait ExtensionFactory {
     fn create(&self) -> anyhow::Result<BoxedExtension>;
 }
 
-pub type BoxedExtension = Box<dyn OakApiNativeExtension + Send + Sync>;
+pub enum Extension {
+    Native(Box<dyn OakApiNativeExtension + Send + Sync>),
+    Uwabi(Box<dyn UwabiExtension + Send + Sync>),
+}
+
+pub type BoxedExtension = Extension;
 pub type BoxedExtensionFactory = Box<dyn ExtensionFactory + Send + Sync>;
 
+/// Trait for implementing an extension which relies on uwabi.
+pub trait UwabiExtension {
+    /// Get the channel handle to address this extension.
+    fn get_channel_handle(&self) -> ChannelHandle;
```

Comment:
Why not an associated const? Or failing, that, an associated method? AFAICT there is nothing that needs `&self` in the implementation of this fn.

Reply:
Ah never mind, at the moment you need to get that back from the instance, since it's not known statically.